pyproject.toml

```diff
@@ -25,7 +24,71 @@ classifiers = [
     "License :: OSI Approved :: MIT License",
     "Operating System :: OS Independent",
 ]
+dynamic = ["optional-dependencies"]
+
+[tool.setuptools.dynamic.optional-dependencies]
+dev = { file = ["requirements-dev.txt"] }
+
+[tool.ruff]
+select = [
+    # pycodestyle
+    "E",
+    # pyflakes
+    "F",
+    # pydocstyle
+    "D",
+    # flake8-bandit
+    "S",
+    # pyupgrade
+    "UP",
+    # pep8-naming
+    "N",
+]
+# Error E501 (Line too long) is ignored because of docstrings.
+ignore = [
+    "S101",
+    "S301",
+    "S311",
+    "D100",
+    "D200",
+    "D203",
+    "D205",
+    "D401",
+    "E501",
+]
+extend-exclude = ["tests/*.py", "setup.py"]
+target-version = "py38"
+force-exclude = true
+
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["E402", "F401", "F403", "F811"]
+
+[tool.ruff.pydocstyle]
+convention = "google"
```

Comment:
@VDuchauffour, in your opinion, is there any practical benefits in using the google format? My personal preference is to go with numpy.

Reply:
No practical benefits in my opinion. I think the preference only depends on habits. We can switch to the Numpy convention if you prefer it.

Reply:
I see. In this case could you change it to numpy? Other than that everything looked great. I think we can merge your PRs tomorrow.

Reply:
Of course. I will do that!